packages/svelte/src/internal/client/proxy.js

```diff
@@ -34,7 +34,11 @@ export const READONLY_SYMBOL = Symbol('readonly');
 export function proxy(value, immutable = true) {
 	if (typeof value === 'object' && value != null && !is_frozen(value)) {
 		if (STATE_SYMBOL in value) {
-			return /** @type {import('./types.js').ProxyMetadata<T>} */ (value[STATE_SYMBOL]).p;
+			const proxy2 = /** @type {import('./types.js').ProxyMetadata<T>} */ (value[STATE_SYMBOL]);
+			// Check that the incoming value is the same proxy that this state symbol was created for:
+			// If someone copies over the state symbol to a new object (using Reflect.ownKeys) the referenced
+			// proxy could be stale and we should not return it.
+			if (proxy2.t === value || proxy2.p === value) return proxy2.p;
```

Comment:
let's avoid `foo2` style var names, it's always a bit of a code smell

```suggestion
			const metadata = /** @type {import('./types.js').ProxyMetadata<T>} */ (value[STATE_SYMBOL]);
			// Check that the incoming value is the same proxy that this state symbol was created for:
			// If someone copies over the state symbol to a new object (using Reflect.ownKeys) the referenced
			// proxy could be stale and we should not return it.
			if (metadata.t === value || metadata.p === value) return metadata.p;
```

Reply:
Ah woops I meant to clean that up but forgot.